test/Common/TestOrchestrations.cs

```diff
@@ -953,6 +953,50 @@ public static async Task<string> RollbackSignalsOnExceptions([OrchestrationTrigg
             return "ok";
         }
 
+        public static async Task<string> HandleUncallableFunctions([OrchestrationTrigger] IDurableOrchestrationContext ctx)
+        {
+            var entityId = ctx.GetInput<EntityId>();
+
+            // even though the entity function cannot be called the entity can still be used as a lock!
+            using (await ctx.LockAsync(entityId))
+            {
+                try
+                {
+                    var result = await ctx.CallEntityAsync<int>(entityId, "get");
+                }
+#if FUNCTIONS_V1
+                catch (Exception e) when (e.Message == "Exception has been thrown by the target of an invocation.")
+#else
+                catch (Exception e) when (e.Message == "Exception of type 'System.Exception' was thrown.")
+#endif
+                {
+                }
+            }
+
+            // signals fail silently
+            ctx.SignalEntity(entityId, "hello");
+
+            // test that the lock is still available
+            using (await ctx.LockAsync(entityId))
+            {
+            }
+
+            // try an uncallable activity
+            try
+            {
+                var result = await ctx.CallActivityAsync<int>(nameof(UnconstructibleClass.UncallableActivity), null);
+            }
+#if FUNCTIONS_V1
+                catch (FunctionFailedException e) when (e.Message == "The activity function 'UncallableActivity' failed: \"Exception has been thrown by the target of an invocation.\". See the function execution logs for additional details.")

```

Comment:
be careful with indentation in your conditional compilation. Make sure that you aren't getting stylecop warnings in either compilation mode.